modules/5_install/install.tf

```diff
@@ -186,7 +187,7 @@ resource "null_resource" "install" {
 
 resource "null_resource" "upgrade" {
     depends_on = [null_resource.install]
-    count      = var.upgrade_image != "" ? 1 : 0
+    count      = var.upgrade_version != "" ? 1 : 0
```

Comment:
May be we want to upgrade only when both `upgrade_version` and `upgrade_channel` are not empty.